### ISSUE TITLE
Question: Why does ARPL employe the difference between the Euclidean distance and dot product as the metric?

### ISSUE THREAD

**Issue by diggerdu** [iCGY96/ARPL#11]
Many thanks for your wonderful work! But the metric is not very intuitive. Do you have some theory analysis or design it in an emprical way.

**Comment by cheng0719** [comment]
> Many thanks for your wonderful work! But the metric is not very intuitive. Do you have some theory analysis or design it in an emprical way.

My understanding is as follows: When treating the original logits of a sample and the reciprocal point as vectors, if their directions are opposite, the dot product will be negative. Therefore, in the code, subtracting the dot product value is intended to encourage the sample and its corresponding class's reciprocal point to be as opposite in direction as possible. This results in larger logits values, leading to a higher probability distribution.
